pkg/controller/hosts_reconciler.go

```diff
@@ -0,0 +1,256 @@
+// SPDX-FileCopyrightText: 2023-present Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package controller
+
+import (
+	"context"
+	"github.com/onosproject/onos-api/go/onos/topo"
+	"github.com/onosproject/topo-discovery/pkg/southbound"
+	"strconv"
+	"sync"
+	"time"
+)
+
+// HostReconciler provides state and context required for host discovery and reconciliation
+type HostReconciler struct {
+	southbound.HostListener
+	hostDiscovery southbound.HostDiscovery
+	topoClient    topo.TopoClient
+	ctx           context.Context
+	lock          sync.RWMutex
+
+	// Map of agent-id to topo object required to resolve link reports to a device
+	agentDevices map[string]*topo.Object
+
+	// Map of agent-id to a list of links that reference that agent ID, but cannot be
```

Comment:
You won't need this, because when you learn from the hosts, it is only from the agent who you are already talking to. This was used for links, where there was the agent on the egress side of the link, which we may not yet have talked to.

Reply:
Addressed